lib/rules/async-server-action.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @fileoverview Require functions with the `use server` directive to be async
+ * @author Jorge Zreik
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+const report = require('../util/report');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const messages = {
+  asyncServerAction: 'Your server action should be async',
+};
+
+/**
+ * Detects a `use server` directive in a given AST node
+ * @param {ASTNode} node The node to search.
+ * @returns {boolean} Whether the node given has a `use server` directive.
+ */
+function hasUseServerDirective(node) {
+  if (node.body.type !== 'BlockStatement') return false;
+
+  const functionBody = node.body.body;
+  if (functionBody.length === 0) return false;
+
+  const potentialDirectiveStatement = functionBody[0];
+  if (potentialDirectiveStatement.type !== 'ExpressionStatement') return false;
+
+  const potentialDirectiveExpression = potentialDirectiveStatement.expression;
+  if (potentialDirectiveExpression.type !== 'Literal') return false;
+
+  return potentialDirectiveExpression.value === 'use server';
```

Comment:
is there a way to use an AST selector string here, like `FunctionDeclaration BlockStatement ExpressionStatement Literal[value="use server"]`, and then this function wouldn't be required at all?

Reply:
Great suggestion! I simplified the rule a ton with a selector.